src/Chapter4.hs

```diff
@@ -629,7 +660,13 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
```

Comment:
I do not really understand the logic for the last of these four cases.
Why shouldn't the last case also produce a `Nothing`? Is it due to some short circuiting of `&&`?
`andM Nothing Nothing 'shouldBe' Nothing`
`andM Nothing (Just True) 'shouldBe' Nothing`
`andM (Just True) Nothing 'shouldBe' Nothing`
`andM (Just False) Nothing 'shouldBe' Just False`